values/src/dimension.rs

```diff
@@ -490,44 +491,170 @@ pub enum Ambiguity {
-        write!(fmt, "<TimeValue>")
+        write!(fmt, "DatetimeValue(form={:?}, grain={:?}, min-grain={:?}, kind={:?})",
+               self.form, self.constraint.grain(), self.constraint.grain_min(), self.datetime_kind)
+
     }
 }
 
-impl TimeValue {
+impl DatetimeValue {
     pub fn is_coarse_grain_smaller_than(&self, grain: Grain) -> bool {
         (self.constraint.coarse_grain_step() as usize) < (grain as usize)
     }
 
     pub fn is_coarse_grain_greater_than(&self, grain: Grain) -> bool {
         (self.constraint.coarse_grain_step() as usize) > (grain as usize)
     }
+
+    pub fn date_form(&self) -> Option<bool> {
+        match self.form {
+            Form::Cycle(grain) => {
+                match grain {
+                    Grain::Year => Some(true),
+                    Grain::Quarter => Some(true),
+                    Grain::Month => Some(true),
+                    Grain::Week => Some(true),
+                    Grain::Day => Some(true),
+                    _ => Some(false),
+                }
+            },
+            Form::Year(_) => Some(true),
+            Form::Month(_) => Some(true),
+            Form::MonthDay(_) => Some(true),
+            Form::YearMonthDay(_) => Some(true),
+            Form::TimeOfDay(_) => Some(false),
+            Form::DayOfWeek { .. } => Some(true),
+            Form::Empty => None,
+            Form::PartOfDay { .. } => Some(false),
+            Form::Meal => Some(false),
+            Form::Celebration => Some(true),
+            Form::PartOfMonth => Some(true),
+            Form::PartOfYear => Some(true),
+            Form::DayOfMonth => Some(true),
+            Form::PartOfForm(_) => None,
+            Form::PartOfWeek => Some(true),
+            Form::Span => None,
+        }
+    }
+
+    pub fn time_form(&self) -> Option<bool> {
+        match self.form {
+            Form::Cycle(grain) => {
+                match grain {
+                    Grain::Hour => Some(true),
+                    Grain::Minute => Some(true),
+                    Grain::Second => Some(true),
+                    _ => Some(false),
+                }
+            },
+            Form::Year(_) => Some(false),
+            Form::Month(_) => Some(false),
+            Form::MonthDay(_) => Some(false),
+            Form::YearMonthDay(_) => Some(false),
+            Form::TimeOfDay(_) => Some(true),
+            Form::DayOfWeek { .. } => Some(false),
+            Form::Empty => None,
+            Form::PartOfDay { .. } => Some(true),
+            Form::Meal => Some(true),
+            Form::Celebration => Some(false),
+            Form::PartOfMonth => Some(false),
+            Form::PartOfYear => Some(false),
+            Form::DayOfMonth => Some(false),
+            Form::PartOfForm(_) => None,
+            Form::PartOfWeek => Some(false),
+            Form::Span => None,
+        }
+    }
+
+    pub fn period_form(&self) -> Option<bool> {
```

Comment:
`is_period_form` would be better as this returns a boolean value.

Reply:
✅